android/app/src/main/java/org/jitsi/meet/MainActivity.java

```diff
@@ -85,16 +113,66 @@ protected boolean extraInitialize() {
 
     @Override
     protected void initialize() {
+        broadcastReceiver = new BroadcastReceiver() {
+            @Override
+            public void onReceive(Context context, Intent intent) {
+                // As new restrictions including server URL are received,
+                // conference should be restarted with new configuration.
+                leave();
+                recreate();
+            }
+        };
+        registerReceiver(broadcastReceiver,
+            new IntentFilter(Intent.ACTION_APPLICATION_RESTRICTIONS_CHANGED));
+
+        resolveRestrictions();
+        setJitsiMeetConferenceDefaultOptions();
+        super.initialize();
+    }
+
+    @Override
+    public void finish() {
+        if (broadcastReceiver != null) {
```

Comment:
Since the broadcastReceiver it is set in the onCreate(), I think we should put this in onDestroy(). If we use Don't keep activities and put the app in background the activity will be destroyed, but not finished and after that onCreate will create a new broadcast receiver, without removing the old one.

Reply:
Isn't it in initialized, not onCreate()?

Reply:
yeah, I meant indirectly in onCreate, since it is there where the initialize method is called.

Reply:
I was honstly thinking those are kinda of same level, pair function initialize/finish.
If not, then maybe both should move explicitly to onCreate/onDestroy?

Reply:
That makes sense to me.

Reply:
Initialize it's just our method in JitsiMeetActivity, called in onCreate, it's not a lifecycle method. It would definitely make sense to have this logic in onCreate/onDestroy, but we have to be careful with what initiliaze does in JitsiMeetActivity and how we call super.onCreate in MainActivity.

Reply:
Yes, clear point. I'm new to this code, so should have reviewed it a bit more carefully.
So, then what is your opinion?
Keep broadcastReceiver creation in initialize and move destruction to onDestroy?
Or create broadcastReceiver in onCreate (then before super or after?)?

Reply:
Since it is not a good practice to call super.onCreate after doing stuff in the overriden method, we would have 2 options: we keep the broadcastReceiver initialisation in initialize and destroy it in onDestroy. Or, return true no matter what in the extraInitialize method of the MainActivity, which will stop JitsiMeetActivy.onCreate from calling initialize, override onCreate in MainActivity, call super.onCreate, then instantiate here the broadcastReceiver+other stuff and then call initialize() from JitsiMeetActivity. I would personally go with the later, but I don't have a strong opinion about it.

Reply:
Both options does not really look intuitive, but I would prefer the simple one, which is #1 in this case, as returning true to skip initialized call seems to be a side effect.